opensquirrel/passes/exporter/cqasmv1_exporter.py

```diff
@@ -35,24 +35,27 @@ def visit_float(self, f: SupportsFloat) -> str:
 
     def visit_measure(self, measure: Measure) -> None:
         qubit_argument = measure.arguments[0].accept(self)  # type: ignore[index]
-        self.cqasmv1_string += f"{measure.name}_z {qubit_argument}\n"
+        self.output += f"{measure.name}_z {qubit_argument}\n"
 
     def visit_init(self, init: Init) -> None:
         qubit_argument = init.arguments[0].accept(self)  # type: ignore[index]
-        self.cqasmv1_string += f"prep_z {qubit_argument}\n"
+        self.output += f"prep_z {qubit_argument}\n"
 
     def visit_reset(self, reset: Reset) -> None:
         qubit_argument = reset.arguments[0].accept(self)  # type: ignore[index]
-        self.cqasmv1_string += f"prep_z {qubit_argument}\n"
+        self.output += f"prep_z {qubit_argument}\n"
 
     def visit_barrier(self, barrier: Barrier) -> None:
-        qubit_argument = barrier.arguments[0].accept(self)  # type: ignore[index]
-        self.cqasmv1_string += f"barrier {qubit_argument}\n"
+        if self.barrier_links and barrier == self.barrier_links[0][-1]:
```

Comment:
Code like this should be more readable. For example, express what you want to do with a function:

```
if is_last_barrier_in_the_current_barriers_group(barrier):
```